kbase-extension/static/kbase/js/widgets/kbaseTabs.js

```diff
@@ -64,167 +64,208 @@ define(['kbwidget', 'bootstrap', 'jquery', 'kbaseDeletePrompt'], (
+        },
+
+        renderDeleteTabButton: function (tab) {
+            return $('<button>')
+                .addClass('btn btn-default btn-xs')
+                .append($('<span>').append('&#x2716;'))
+                .css('padding', '0px')
+                .css('width', '12px')
+                .css('height', '12px')
+                .css('margin-left', '10px')
+                .css('font-size', '10px')
+                .css('margin-bottom', '3px')
+                .css('outline', 'none')
+                .css('border', '0')
+                .bind(
+                    'click',
+                    $.proxy(function (e) {
+                        e.preventDefault();
+                        e.stopPropagation();
+                        if (typeof tab.deleteCallback !== 'undefined') {
+                            tab.deleteCallback(tab.tab);
+                        } else if (typeof tab.confirmDelete === 'undefined') {
+                            // In this case, there is no confirmDelete set on the tab,
+                            // so we default to the global behavior.
+                            if (
+                                typeof this.options.confirmDelete === 'undefined' ||
+                                this.options.confirmDelete
+                            ) {
+                                this.deletePrompt(tab.tab);
+                            } else {
+                                this.removeTab(tab.tab);
+                            }
+                        } else if (tab.confirmDelete) {
+                            this.deletePrompt(tab.tab);
+                        } else {
+                            this.removeTab(tab.tab);
+                        }
                     }, this)
                 );
+        },
+
+        renderTabPanel: function (tab, $tabPanel) {
+            // This may be called before the tab is added
+            $tabPanel = $tabPanel || this.data('tabs')[tab.tab];
+            if ($tabPanel.hasContent) {
+                return;
+            }
+            if (tab.content) {
+                $tabPanel.append(tab.content);
+                $tabPanel.hasContent = true;
+            } else if (tab.showContentCallback) {
+                try {
+                    $tabPanel.html(tab.showContentCallback($tabPanel));
+                } catch (ex) {
+                    $tabPanel.html($ErrorMessage(ex));
+                }
+                if (!tab.dynamicContent) {
+                    $tabPanel.hasContent = true;
+                }
             }
         },
 
         addTab: function (tab) {
-            if (tab.canDelete == undefined) {
+            if (typeof tab.canDelete === 'undefined') {
                 tab.canDelete = this.options.canDelete;
             }
 
-            const $tab = $('<div role="tabpanel"></div>').addClass('tab-pane fade');
-
-            $tab.hasContent = false;
+            const $tabPanel = $('<div role="tabpanel"></div>').addClass('tab-pane fade');
```

Comment:
chose a better name; with all the "tab"s floating around (including "tab.tab"!), it is important to be more specific in naming things.